### ISSUE TITLE
Use different interpolation for inverted random resampling transforms

### ISSUE THREAD

**Issue by fepegar** [fepegar/torchio#410]
**🚀 Feature**

It would be good to be able to select one interpolation for the forward resampling transform and a different one for the inverted one.

**Motivation**

Test-time augmentation: a random resampling transform T (say `RandomAffine`) is applied to image X with interpolation I1 to obtain Xt. To minimize the quality loss during resampling, we set I1 = B-spline interpolation. Logits are softmaxed to obtain a probability map Pt.

Two scenarios:
1. We apply T^(-1) to Pt to get P, the probabilities in the original space. If we use I1, then it is possible that not all values in P are in [0, 1]. In this case, I would like I2 != I1. For example, I would like I2 to be a linear interpolation.
2. We have a very large P. For example, in the [inference notebook](https://github.com/fepegar/torchio/blob/master/examples/README.md#inference), the shape of P is something like `(1, 180, 200, 200, 200)`. Instead of resampling a volume of this size, which is probably not possible with a normal RAM, we compute the label map using argmax and resample back using nearest-neighbor interpolation.

**Pitch**

I suppose the second case could be solved if a `LabelMap` is used before inverting the transform. For the first case, there should be a way of specifying that we want a different interpolation for the inverse transform. I can think of two possible solutions:
1. A kwarg in all resampling transforms
2. A kwarg in `Subject.apply_inverse_transform`

**Alternatives**

In our two scenarios, use only linear for 1) and only nearest-neighbor for 2).

**Comment by romainVala** [comment]
Hi
just a note, to understand how you proceed, for the 1) ?
do you add the Pt to the subject (as a scalar image). if you add it as a labelMap, then you can not change the interpolation ...

About an alternatives, can you just change the interpolation method "on the fly" after getting the history transform ?

**Comment by fepegar** [comment]
> just a note, to understand how you proceed, for the 1) ?
do you add the Pt to the subject (as a scalar image). if you add it as a labelMap, then you can not change the interpolation ...

That's right. For now, I either add a LabelMap if I want NN interpolation (as in [this notebook](https://github.com/fepegar/torchio-notebooks/blob/main/notebooks/Brain_parcellation_with_TorchIO_and_HighRes3DNet.ipynb)) or a ScalarImage if I want e.g. linear.

> About an alternatives, can you just change the interpolation method "on the fly" after getting the history transform ?

Yes, you're right. I just did that in a script. But I think there should be a higher-level way of doing this.

**Comment by fepegar** [comment]
FYI I got owned again by my own code. I was modifying the last transform in `Subject.history`, but this is not the way because `history` returns a copy generated on the fly from `Subject.applied_transforms`. What I'm doing for now is get the composed inverse transform first with `Subject.get_inverse_transform()` and modify its `transforms` attribute. Very hacky...

**Comment by romainVala** [comment]
thanks for sharing the example
I do not see the` Subject.get_inverse_transform() ` from your notebook, you did not modify it or I miss it ?

**Comment by fepegar** [comment]
I don't need it in the notebook, because I want NN for the inverse transform. The example was for scenario 1.
